packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -444,36 +444,33 @@ class DevFS {
     assert(trackWidgetCreation != null);
     assert(generator != null);
 
+    // Update modified files
+    final String assetBuildDirPrefix = _asUriPath(getAssetBuildDirectory());
+    final Map<Uri, DevFSContent> dirtyEntries = <Uri, DevFSContent>{};
+
+    int syncedBytes = 0;
     if (bundle != null) {
       printTrace('Scanning asset files');
       // We write the assets into the AssetBundle working dir so that they
       // are in the same location in DevFS and the iOS simulator.
       final String assetDirectory = getAssetBuildDirectory();
       bundle.entries.forEach((String archivePath, DevFSContent content) {
         final Uri deviceUri = fs.path.toUri(fs.path.join(assetDirectory, archivePath));
-        _entries[deviceUri] = content;
+        if (deviceUri.path.startsWith(assetBuildDirPrefix)) {
+          archivePath = deviceUri.path.substring(assetBuildDirPrefix.length);
+        }
+        // When doing full restart, copy content so that isModified does not
+        // reset last check timestamp because we want to report all modified
+        // files to incremental compiler next time user does hot reload.
+        if ((_lastUpdate != null && content.isModifiedAfter(_lastUpdate)) || (bundleFirstUpload && archivePath != null)) {
```

Comment:
this check used to involve `bundleDirty`(which is now unused) and the fact that DevFSContent managed it's `isModified` state. Now it seems we just rely on keeping `_lastUpdate` here and comparing modification date against it. Why this change?

Reply:
Since _entries previously contained both assets and source files, it was necessary to separate the check to update asset paths. Now that we only ever upload assets + dill, these checks can be combined. In this case, I don't think there is any need for bundleDirty, since we can use the bundle itself as the source of truth.

Actually, having fixed the bundle bug I can probably return this to using isModified instead.